mockwebserver/src/test/java/okhttp3/mockwebserver/RecordedRequestTest.java

```diff
@@ -32,6 +31,16 @@ private FakeSocket(InetAddress inetAddress, int port) {
     }
   }
 
+  @Test public void testIPv4Proxy() throws UnknownHostException {
+    Socket socket =
+        new FakeSocket(InetAddress.getByAddress("127.0.0.1", new byte[] { 127, 0, 0, 1 }), 80);
+
+    RecordedRequest request = new RecordedRequest("CONNECT android.com:443 HTTP/1.1", headers,
+        Collections.<Integer>emptyList(), 0, new Buffer(), 0, socket);
+
+    assertEquals("http://127.0.0.1/android.com:443", request.getRequestUrl().toString());
```

Comment:
I don't think this is right. I would prefer `http://android.com/` here. It's incomplete, but it's all we know about the URL.

We might want to change the parameter in request line from `path` to something that applies in both cases.